src/waveform/renderers/waveformmarkset.h

```diff
@@ -22,10 +22,13 @@ class WaveformMarkSet {
     WaveformMark& operator[] (int i) { return m_marks[i]; }
 
     const WaveformMark& getDefaultMark() const { return m_defaultMark;}
+    // hotCue must be valid (>= 0)
```

Comment:
This assumption should be checked at least by an assertion in the implementation:
DEBUG_ASSERT(hotCue >= 0);

There is not only a lower, but also an upper bound when accessing the members of m_marks